### ISSUE TITLE
vol.toJSON() does not contain stats of a file

### ISSUE THREAD

**Issue by marco-a** [streamich/memfs#562]
First of all, great library!

I wanted to use it for unit testing a part of my application.

Sadly it seems like `vol.toJSON()` does not export file stats (such as `uid`, `gid` etc.).

I had a quick glance over the source code and it looks like these are returned _somewhere_:

https://github.com/streamich/memfs/blob/6842252515388164e2528362b075025ee6caaf1f/src/node.ts#L214-L229

```javascript
const vol = memfs.Volume.fromJSON({ '/foo': 'bar' });
vol.readFileSync('/foo'); // bar
vol.chownSync("/foo", 200, 200)
// logs 200 as expected
console.log(vol.statSync("/foo").uid)
// does not export set uid & gid
console.log(vol.toJSON())
```

Is this intentional ?

**Comment by G-Rath** [comment]
Yes, because `toJSON` is meant to be the opposite of `fromJSON`, which doesn't support setting stats of a file.

**Comment by takase1121** [comment]
Is it possible to make it export the stat alongside the content? This'll allow users to preserve the entire filesystem (including metadata) in localStorage or maybe IndexedDB without messing up the mtimes.
